ConfidentialClientTokenCache/Program.cs

```diff
@@ -0,0 +1,189 @@
+            var result = await app.AcquireTokenForClient(scopes)
+                .ExecuteAsync();
+            Console.WriteLine(result.AuthenticationResultMetadata.TokenSource);
+
+            result = await app.AcquireTokenForClient(scopes)
+                .ExecuteAsync();
+            Console.WriteLine(result.AuthenticationResultMetadata.TokenSource);
+
+        }
+
+        /// <summary>
+        /// Creates a token cache (implementation of your choice)
+        /// </summary>
+        /// <param name="cacheImplementation">implementation for the token cache</param>
+        /// <returns>An Msal Token cache provider</returns>
+        private static IMsalTokenCacheProvider CreateTokenCache(CacheImplementationDemo cacheImplementation=CacheImplementationDemo.InMemory)
+        {
+            IServiceCollection services = new ServiceCollection();
+
+            // (Simulates the configuration, could be a IConfiguration or anything)
+            Dictionary<string, string> Configuration = new Dictionary<string, string>();
+
+            switch (cacheImplementation)
+            {
+                case CacheImplementationDemo.InMemory:
+                    // In memory token cache
+                    // In net472, requires to reference Microsoft.Extensions.Caching.Memory
+                    services.AddInMemoryTokenCaches();
+                    break;
+
+                case CacheImplementationDemo.DistributedMemory:
+                    // In memory distributed token cache
+                    // In net472, requires to reference Microsoft.Extensions.Caching.Memory
+                    services.AddDistributedTokenCaches();
+                    services.AddDistributedMemoryCache();
+                    break;
+
+                case CacheImplementationDemo.DistributedSqlServer:
+                    // SQL Server token cache
+                    // Requires to reference Microsoft.Extensions.Caching.SqlServer
+                    services.AddDistributedTokenCaches();
+                    services.AddDistributedSqlServerCache(options =>
+                    {
+                        options.ConnectionString = @"Data Source=(localdb)\MSSQLLocalDB;Initial Catalog=TestCache;Integrated Security=True;Connect Timeout=30;Encrypt=False;TrustServerCertificate=False;ApplicationIntent=ReadWrite;MultiSubnetFailover=False";
+                        options.SchemaName = "dbo";
+                        options.TableName = "TestCache";
+                    });
+                    break;
+
+                case CacheImplementationDemo.StackExchangeRedis:
+                    // Redis token cache
+                    // Requires to reference Microsoft.Extensions.Caching.StackExchangeRedis
+                    services.AddDistributedTokenCaches();
+                    services.AddStackExchangeRedisCache(options =>
+                    {
+                        options.Configuration = "localhost";
+                        options.InstanceName = "SampleInstance";
```

Comment:
should these also be injected from the Configuration (same above) - and having the config object initialized with these values?

Reply:
I've added an issue to do it later: https://github.com/Azure-Samples/active-directory-dotnet-v1-to-v2/issues/20
proposing to postpone it after the release (it's more urgent / important to release MSAL.NET)